src/sim/include/AbstractDriver.h

```diff
@@ -1244,7 +1243,7 @@ namespace MFM
 		      "\"symbol\":\"%s\", "
 		      "\"name\":\"%s\", "
 		      "\"argb\":%d, "
-		      "\"data_string\":\"%s\"}"
+		      "\"data_members\":%s}"
```

Comment:
If this Printf was broken into two parts -- the part up through "data_string":, and the part after the %s there -- then the OString buff could be avoided entirely.  (Admittedly it was always gross.)  fs.Printf(FRONT_PART); then uelt->Print(ucr,fs,*atom,ETC); then fs.Printf(BACK_PART).  Passing fs to uelt->Print instead of buff.

If we're pretty confident OString2048 will do the job, given how much expansion we might get printing one atom in JSON, maybe not such an issue.

Reply:
I could begin to speculate the max character length of the data members object, but chances are I'd be wrong.
Funny enough, I left a todo in the latest commit about converting the event_layer_atoms and base_layer_atoms lists into char buffers (up to the number of sites in the grid times the max length of an atom long!). I didn't do it because I'd want to use one of these OverflowableCharBufferByteSync templates with a buffsize that scales with the grid size (though I imagine this might kill compile times so it may be not worth it for that reason alone).
The problems were I was hoping to solve with this approach were: 
1. to avoid excessive fopen/fclose operations by caching the whole JSON document in memory before streaming the whole thing to disk in one go. I mean, I'm running a rather large tmpfs for my /tmp, but supposing someone specifies their output to be written to, e.g. a 5500 rpm HDD, I'm wondering if that would unnecessarily increase iowait on the system running the simulation (and the frustration of the programmer using the simulator). And,
2. so that I can grab non-empty atoms from the event layer and base layer in one pass.

I tried digging into the FileByteSync to see if this problem is already solved, but I didn't get to the bottom of it.
I think maybe I'm trying to optimize too much for an unlikely scenario. 

tldr; if you think I should drop the buffers altogether, I can pass the FileByteSync object instead.